moveit_simple/src/moveit_simple.cpp

```diff
@@ -63,8 +64,54 @@ Robot::Robot(const ros::NodeHandle & nh, const std::string &robot_description,
 
 
 
+Robot::Robot(const ros::NodeHandle & nh,
+                      const std::string &robot_description,
+                      const std::string &group_name,
+                      const std::string &kinematics_base_link,
+                      const std::string &kinematics_tip_link):
+  tf_buffer_(),
+  tf_listener_(tf_buffer_),
+  nh_(nh),
+  params_(ros::NodeHandle("~/moveit_simple")),
+  speed_modifier_(1.0),
+  dynamic_reconfig_server_(ros::NodeHandle("~/moveit_simple"))
+{
+  ROS_INFO_STREAM("Loading MoveIt objects based on, robot description: " << robot_description
+                  << ", group name: " << group_name);
+  robot_model_loader_.reset(new robot_model_loader::RobotModelLoader
+                            (robot_description));
+  robot_model_ptr_ = robot_model_loader_->getModel();
+  virtual_robot_state_.reset(new moveit::core::RobotState(robot_model_ptr_));
+  virtual_robot_state_->setToDefaultValues();
+  planning_scene_.reset(new planning_scene::PlanningScene(robot_model_ptr_));
+  joint_group_ = robot_model_ptr_->getJointModelGroup(group_name);
+  ROS_INFO_STREAM("Calculating all positions assuming, root: " << robot_model_ptr_->getRootLinkName()
+                  << ", and tool: " << robot_model_ptr_->getLinkModelNames());
+  ROS_INFO_STREAM("MoveIt object loaded");
+
+  virtual_visual_tools_.reset(new moveit_visual_tools::MoveItVisualTools(
+                        robot_model_ptr_->getRootLinkName(),
+                        nh_.getNamespace() + "/rviz_visual_tools",
+                        robot_model_ptr_));
+  virtual_visual_tools_->loadRobotStatePub(nh_.getNamespace()
+                                    + "/display_robot_state");
+
+  // Dynamic Reconfigure Parameters with rosparam_handler
+  params_.fromParamServer();
+  dynamic_reconfig_server_.setCallback(boost::bind(&Robot::reconfigureRequest, this, _1, _2));
+
+  kinematics_base_link_override = kinematics_base_link;
```

Comment:
Instead cache these values as either overrides or the `joint_group_->getSolverInstance()->getTipFrame();`, etc...